### ISSUE TITLE
Support for ESP-32

### ISSUE THREAD

**Issue by alpha815** [ncmreynolds/pimoroniTrackball#3]
While i was trying to hookup this breakout board to esp-32 and loaded basic example code, it seems that RGB values in setup are being set by esp-32, but when i move the trackball it does not appear to update the left, right, up and down position. then again i tested it on Arduino Uno it worked fine.

Arduino ESP-32 Core Version: 2.0.3
ESP-32 board tested on ESP-32 TINY-PICO.

So it seems for some reasons support for esp-32 is broken.

Serial monitor output with esp-32.
![output](https://user-images.githubusercontent.com/56131964/188432082-2f7f0ad6-a4bb-4aea-881d-f40e98df64f8.jpg)
Serial monitor output with Arduino.
![Uno](https://user-images.githubusercontent.com/56131964/188432206-a153aa80-68e8-41a2-bacf-bc9419a19833.jpg)

Compiler Warning;
```
D:\Arduino\libraries\pimoroniTrackball-0.1.2\src\pimoroniTrackball.cpp: In member function 'bool pimoroniTrackball::isConnected()':
D:\Arduino\libraries\pimoroniTrackball-0.1.2\src\pimoroniTrackball.cpp:37:51: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second:
   _i2cPort->requestFrom(_trackballAddress, 2, true);
```
                                                   ^

**Comment by ncmreynolds** [comment]
Hi, thanks for spotting this issue I only tested with AVR and ESP8266. I've been able to replicate it on ESP32-S2 and am looking at it now.

**Comment by ncmreynolds** [comment]
This should be fixed in release 0.1.3. Which I've just packaged up and will appear in the Arduino Library Manager soon.
